docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -0,0 +1,571 @@
+kind: Logstash
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRefs:
+  - name: quickstart
+    clusterName: qs
+  config: <1>
+    pipeline.workers: 4  
+    log.level: debug
+----
+<1> Customize Logstash configuration using `logstash.yml` settings here
+
+
+Alternatively, it can be provided through a Secret specified in the `spec.configRef` section. The Secret must have an `logstash.yml` entry with this configuration:
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRefs:
+  - name: quickstart
+    clusterName: qs
+  configRef:
+    secretName: quickstart-config
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: quickstart-config
+stringData:
+  logstash.yml: |-
+    pipeline.workers: 4
+    log.level: debug
+----
+
+
+[id="{p}-logstash-pipelines"]
+=== Configuring Logstash Pipelines
+
+Logstash pipelines (equivalent to `pipelines.yml`) are defined in the `spec.pipelines` section:
+
+[source,yaml,subs="attributes,+macros,callouts"]
+----
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRefs:
+    - clusterName: qs
+      name: quickstart
+  pipelines:
+    - pipeline.id: main
+      config.string: |
+        input {
+          beats {
+            port => 5044
+          }
+        }
+        output {
+          elasticsearch {
+            hosts => [ "${QS_ES_HOSTS}" ]
+            user => "${QS_ES_USER}"
+            password => "${QS_ES_PASSWORD}"
+            cacert => "${QS_ES_SSL_CERTIFICATE_AUTHORITY}"
+          }
+        }
+----
+
+Alternatively, it can be provided through a Secret specified in the `spec.pipelinesRef` element. The Secret must have a `logstash.yml` entry with this configuration:
+[source,yaml,subs="attributes,+macros"]
+----
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: quickstart
+spec:
+  version: {version}
+  count: 1
+  elasticsearchRefs:
+    - clusterName: qs
+      name: quickstart
+  pipelinesRef:
+    secretName: quickstart-pipeline
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: quickstart-pipeline
+stringData:
+  pipelines.yml: |-
+    - pipeline.id: main
+      config.string: |
+        input {
+          beats {
+            port => 5044
+          }
+        }
+        output {
+          elasticsearch {
+            hosts => [ "${QS_ES_HOSTS}" ]
+            user => "${QS_ES_USER}"
+            password => "${QS_ES_PASSWORD}"
+            cacert => "${QS_ES_SSL_CERTIFICATE_AUTHORITY}"
+          }
+        }
+
+----
+
+Logstash on ECK will support all options present in `pipelines.yml`, including settings to update the number of workers, and
+ the size of the batch that the pipeline will process. This also includes using `path.config` to point to volumes
```

Comment:
I want to point out `path.config` specifically - it provides another way to store pipeline configurations, rather than inline in the `pipeline`, or the secret pointed to in the `pipelineRef` - it's a way to have more complex pipelines than is possible via `pipeline` or `pipelineRef` due to limitations on k8s `Secret` and `ConfigMap` definitions